generate/templates/manual/repository/get_remotes.cc

```diff
@@ -74,7 +74,6 @@ void GitRepository::GetRemotesWorker::Execute()
       }
 
       git_strarray_free(&remote_names);
-      git_repository_free(repo);
```

Comment:
This free should be fine unless I'm missing something. `repo` was opened inside this function, `baton->repo` was opened outside this function.

It actually should probably be freed in the above returns as well...

Reply:
It looks like opening a new repo was to reduce the amount of time we need to lock (just lock while getting the repo path from the repo object vs. lock for the entire function).

You can see in `get_references.cc` in `AcquireLocks()` we pass `baton->repo` to the `LockMaster` so it will be locked while `Execute()` is running vs. in `get_remotes.cc` it isn't, and instead a new `LockMaster` is constructed just while we're opening a new repo instance (this could actually be shortened to just getting the path).

I'd probably just leave the free as it was before and call it good so we don't worry about introducing any new bugs, but it's up to you. If you do make any changes, I'd probably lean towards the more granular locking approach.